Fix #75 SQL scripts without Views/Functions and cleanup migration

antoineatrhea opened the pull request:
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WebServices-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WebServices-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
<!-- A description of the changes proposed in the pull-request -->
Fix #75 

- SQL scripts generated with DotLiquid
- Dao missed some column on SELECT query when a class is contained but is not directly contained (superclass is contained)
- SQL Scripts do not contains any views/data function anymore 
- Migration Script that removes all existing VIEWS/Data function

Remarks : SqlMetadataHelper class may not be necessary but could be useful later, for any SELECT query that may need to be handcoded
<!-- Thanks for contributing to COMET Web Services! -->

lxatrhea reviewed (commented):
Why did the "*_20210109*" Migrations change?

antoineatrhea commented:
@lxatrhea I moved old migrations (with views) onto another folder and just removed views/function from there. I did not change name of migration to not rerun them if already applied

samatrhea commented on CDP4Orm/MigrationScript/EngineeringModel_20210109_6_4_0_0_Model_Updates.sql:
```diff
@@ -1,4 +1,4 @@
-﻿-- Create table for class LogEntryChangelogItem (which derives from: Thing)
+-- Create table for class LogEntryChangelogItem (which derives from: Thing)
```
am i to understand that you are changing existing migrations?

alexatrhea replied:
no, i think he copied the old ones unmodified to Old_WithViews folder and made a new set here by modifying the old ones. However @antoineatrhea I also don't get exactly why this is needed?

antoineatrhea replied:
I kept the old ones to have a traceability. But it's required to still have those migrations since the structure is on version 1.1.27, so we have to apply model updates migration. However, I can not keep the exact same as before since we don't have VIEWS and Data functions anymore. I added a new migration that cleans every VIEWS and Data functions to be sure to not have of them if we have an old database

samatrhea commented on CDP4Orm/MigrationScript/Iteration_20210109_6_5_0_0_Model_Updates.sql:
```diff
@@ -1,4 +1,4 @@
-﻿-- OrganizationalParticipant is a collection property (many to many) of class ElementDefinition: [0..*]-[1..1]
+-- OrganizationalParticipant is a collection property (many to many) of class ElementDefinition: [0..*]-[1..1]
```
am i to understand that you are changing existing migrations?

samatrhea commented on CometServer/Helpers/Cdp4TransactionManager.cs:
```diff
@@ -418,11 +418,12 @@ private static void ApplyIterationContext(NpgsqlTransaction transaction, string
             }
             
             var sqlBuilder = new StringBuilder();
```
here i would like an explanation in a conversation

samatrhea commented on CDP4Orm/AutoGenOrmHelpers/DaoResolver.cs:
```diff
@@ -0,0 +1,226 @@
+        /// Initializes a new <see cref="DaoResolver" />
+        /// </summary>
+        /// <param name="serviceProvider">The injected <see cref="IServiceProvider" /></param>
+        public DaoResolver(IServiceProvider serviceProvider)
```
using serviceprovider.GetService is an anti-pattern https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines

i propose that instead we inject them as properties these services and that there is a helper function that returns the value of the property

antoineatrhea replied:
Should we inject them with a Lazy process? otherwise, at anytime we request DaoResolver, it would also initialize all Dao, no? With the IServiceProvider, we only request them when invoking the method?

samatrhea commented on CDP4Orm/Dao/Resolve/ContainerDao.cs:
```diff
@@ -207,18 +216,20 @@ public class ContainerDao : IContainerDao
 
+            var dao = this.DaoResolver.QueryDaoByTypeName(typeName);
+            
             sqlBuilder.Append(
```
may i request we refactor this so it becomes more readable. using the numbered arguments and a combi of string builder and string.format can be improved

antoineatrhea replied:
refactored

samatrhea commented on CDP4Orm/Dao/Resolve/ResolveDao.cs:
```diff
@@ -86,8 +86,8 @@ private static IEnumerable<ResolveInfo> ReadSiteDirectoryThing(NpgsqlTransaction
             sqlBuilder.Append(
-                "SELECT \"ValueTypeSet\"-> 'ClassKind' as \"TypeInfo\", \"Iid\"").Append(
-                " FROM \"{0}\".\"Thing_View\"").Append(
+                "SELECT \"ValueTypeDictionary\"-> 'ClassKind' as \"TypeInfo\", \"Iid\"").Append(
```
may i request we refactor this so it becomes more readable. using the numbered arguments and a combi of string builder and string.format can be improved

antoineatrhea replied:
refactored

alexatrhea reviewed (commented):
yeah maybe we can have a quick chat on the migrations to understand.
